examples/replay-buffers/catframes-in-buffer.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch
+from torchrl.data import LazyTensorStorage, ReplayBuffer
+from torchrl.envs import (
+    CatFrames,
+    Compose,
+    DMControlEnv,
+    StepCounter,
+    ToTensorImage,
+    TransformedEnv,
+    UnsqueezeTransform,
+)
+
+# Number of frames to stack together
+frame_stack = 4
+# Dimension along which the stack should occur
+stack_dim = -4
+# Max size of the buffer
+max_size = 100_000
+# Batch size of the replay buffer
+training_batch_size = 32
+
+seed = 123
+
+
+def main():
+    env = TransformedEnv(
+        DMControlEnv(
+            env_name="cartpole",
+            task_name="balance",
+            device="cpu",
+            from_pixels=True,
+            pixels_only=True,
+        ),
+        Compose(
+            ToTensorImage(
+                from_int=True,
+                dtype=torch.float32,
+                in_keys=["pixels"],
+                out_keys=["pixels_trsf"],
+                shape_tolerant=True,
+            ),
+            UnsqueezeTransform(
+                dim=stack_dim, in_keys=["pixels_trsf"], out_keys=["pixels_trsf"]
+            ),
+            CatFrames(
+                N=frame_stack,
+                dim=stack_dim,
+                in_keys=["pixels_trsf"],
+                out_keys=["pixels_trsf"],
+            ),
+            StepCounter(),
+        ),
+    )
+    env.set_seed(seed)
+
+    sampler = None
+    transform = None
+
+    def collect_transform_sampler(module):
+        nonlocal sampler
+        nonlocal transform
+        if isinstance(module, CatFrames):
+            transform, sampler = module.make_rb_transform_and_sampler(
+                batch_size=training_batch_size,
+                traj_key=("collector", "traj_ids"),
+                strict_length=True,
+            )
+
+    env.apply(collect_transform_sampler)
+
+    rb_transforms = Compose(
+        ToTensorImage(
+            from_int=True,
+            dtype=torch.float32,
+            in_keys=["pixels", ("next", "pixels")],
+            out_keys=["pixels_trsf", ("next", "pixels_trsf")],
+            shape_tolerant=True,
+        ),  # C W' H' -> C W' H' (unchanged due to shape_tolerant)
+        UnsqueezeTransform(
+            dim=stack_dim,
+            in_keys=["pixels_trsf", ("next", "pixels_trsf")],
+            out_keys=["pixels_trsf", ("next", "pixels_trsf")],
+        ),  # 1 C W' H'
+        transform,
+    )
+
+    rb = ReplayBuffer(
+        storage=LazyTensorStorage(max_size=max_size, device="cpu"),
+        sampler=sampler,
+        batch_size=training_batch_size,
+        transform=rb_transforms,
+    )
+
+    data = env.rollout(1000, break_when_any_done=False)
```

Comment:
I think we should perform 2k steps to be guaranteed to have 2 trajs (max is 1000 for DMC).  
This will allow the addition of assertions that could be used in the doc to explain what is going on further (eg: As you can see, we check that the stacked frames for the frame at index 0 in the sampled batch is actually ...)   
OR  
Keep using 1000 in the example but for the unit tests I would make sure to have a test that ensures that CatFrames did not simply use the previous frame when that previous frame did not belong to the same traj.